packages/flutter/lib/src/widgets/spell_check.dart

```diff
@@ -0,0 +1,345 @@
+  /// for spell check.
+  const SpellCheckConfiguration({
+    this.spellCheckService,
+    this.misspelledTextStyle,
+  }) : _spellCheckEnabled = true;
+
+  /// Creates a configuration that disables spell check.
+  const SpellCheckConfiguration.disabled()
+    : _spellCheckEnabled = false,
+       spellCheckService = null,
+       misspelledTextStyle = null;
+
+  /// The service used to fetch spell check results for text input.
+  final SpellCheckService? spellCheckService;
+
+  /// Style used to indicate misspelled words.
+  ///
+  /// This is nullable to allow style-specific wrappers of [EditableText]
+  /// to infer this, but this must be specified if this configuration is
+  /// provided directly to [EditableText] or its construction will fail with an
+  /// assertion error.
+  final TextStyle? misspelledTextStyle;
+
+  final bool _spellCheckEnabled;
+
+  /// Whether or not the configuration should enable or disable spell check.
+  bool get spellCheckEnabled => _spellCheckEnabled;
+
+  /// Returns a copy of the current [SpellCheckConfiguration] instance with
+  /// specified overrides.
+  SpellCheckConfiguration copyWith({
+    SpellCheckService? spellCheckService,
+    TextStyle? misspelledTextStyle}) {
+    if (!_spellCheckEnabled) {
+      // A new configuration should be constructed to enable spell check.
+      return const SpellCheckConfiguration.disabled();
+    }
+
+    return SpellCheckConfiguration(
+      spellCheckService: spellCheckService ?? this.spellCheckService,
+      misspelledTextStyle: misspelledTextStyle ?? this.misspelledTextStyle,
+    );
+  }
+
+  @override
+  String toString() {
+    return '''
+  spell check enabled   : $_spellCheckEnabled
+  spell check service   : $spellCheckService
+  misspelled text style : $misspelledTextStyle
+'''
+        .trim();
+  }
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) {
+        return true;
+    }
+
+    return other is SpellCheckConfiguration
+      && other.spellCheckService == spellCheckService
+      && other.misspelledTextStyle == misspelledTextStyle
+      && other._spellCheckEnabled == _spellCheckEnabled;
+  }
+
+  @override
+  int get hashCode => Object.hash(spellCheckService, misspelledTextStyle, _spellCheckEnabled);
+}
+
+/// Determines how misspelled words are indicated in text input and how
+/// replacement suggestions for misspelled words are displayed via menu.
+///
+/// Any widget may use this handler to build a [TextSpan] tree with
+/// [SpellCheckResults] by calling
+/// `SpellCheckSuggestionsHandler.buildTextSpanWithSpellCheckSuggestions(...)`.
+///
+/// See also:
+///
+///  * [EditableText], which uses this handler to display spell check results
+///    if spell check is enabled.
+class SpellCheckSuggestionsHandler {
+  /// Creates a handler to use for spell checking text input based on the
+  /// provided platform.
+  SpellCheckSuggestionsHandler();
```

Comment:
Without this, the linter throws `avoid_classes_with_only_static_members`. I played around with moving this code completely to `EditableTextState` but it felt wonky, since it would be harder for users to access these methods (since keeping them static would not make sense anymore) and it would also be weirder to test. Thought about moving it to one of the `SpellCheckConfiguration` or `SpellCheckResults` classes but neither seemed fitting. Also, I think that moving this code out of this class would further commit us to not allowing this handler to be configurable, and I don't believe that is desirable.

Reply:
What if you get rid of the class and just make these top level functions?  Kind of like [showMenu](https://master-api.flutter.dev/flutter/material/showMenu.html) I guess.

What are you goals for making this configurable as you say?  Currently buildTextSpanWithSpellCheckSuggestions can be reused in places other than EditableText (maybe like a custom text editor?) but EditableText's usage of it can't be configured.

Reply:
I think making them top level functions makes sense. I actually was pro-moving them out of their own class 😅  I think I just overthought the impact of moving the methods out of their own class.